diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/DiktatRule.kt

```diff
@@ -0,0 +1,60 @@
+package org.cqfn.diktat.ruleset.rules
+
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.common.config.rules.isRuleEnabled
+import org.cqfn.diktat.ruleset.constants.EmitType
+import org.cqfn.diktat.ruleset.utils.log
+
+import com.pinterest.ktlint.core.Rule
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+private typealias DiktatConfigRule = org.cqfn.diktat.common.config.rules.Rule
+
+/**
+ * This is a wrapper around Ktlint Rule
+ *
+ * @param id id of the rule
+ * @property configRules all rules from configuration
+ * @property rules warnings that are used in the rule's code
+ */
+@Suppress("TooGenericExceptionCaught")
+abstract class DiktatRule(id: String,
+                          val configRules: List<RulesConfig>,
+                          val rules: List<DiktatConfigRule>) : Rule(id) {
+    /**
+     * Default value is false
+     */
+    var isFixMode: Boolean = false
+
+    /**
+     * Will be initialized in visit
+     */
+    lateinit var emitWarn: EmitType
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: EmitType) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        if (areRulesEnabled()) {
+            return
+        } else {
+            try {
+                logic(node)
+            } catch (internalError: Throwable) {
```

Comment:
damn, catching all exceptions? I do not know if it is a good idea, @petertrr think about this here

Reply:
I mean, isn't this what we were trying to achieve? Otherwise these exceptions are caught by ktlint and they just log a warning as well.

Reply:
I am saying that simply doing catch (Exception e) is a bad practice. May be there are different types of exceptions in ktlint to handle BEFORE we will catch all remaining exeptions

Reply:
Everything that is thrown from `rule.visit` is catched in [Ktlint.kt with `catch (e: Exception)`](https://github.com/pinterest/ktlint/blob/65d50c844e2f8d6e34338f8945f6f4eec4522e21/ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/KtLint.kt#L107). So their idea is that the visitor either handles all exceptions, or they are considered internal errors.